cekit/descriptor/packages.py

```diff
@@ -111,5 +117,18 @@ def _get_repo_url(self, descriptor):
     def fetch(self, target_dir):
         if not os.path.exists(target_dir):
                 os.makedirs(target_dir)
-        Resource({'url': self._descriptor['url']['repository']}) \
-            .copy(os.path.join(target_dir, self._descriptor['filename']))
+        if self._repofile:
+            Resource({'url': self._repofile}) \
+                .copy(os.path.join(target_dir, self._descriptor['filename']))
+        else:
+            config = configparser.ConfigParser()
+            sectionName = self._descriptor['name']
+            config.add_section(sectionName);
+            config.set(sectionName, 'name', sectionName);
+            config.set(sectionName, 'baseurl', self._descriptor['url']['repository'])
+            config.set(sectionName, 'enabled', 1)
+            if 'gpg' in self._descriptor['url']:
+                config.set(sectionName, 'gpgkey', self._descriptor['url']['gpg'])
```

Comment:
why have gpg url and do not embed it in descriptor too?

Reply:
It does get embedded in the descriptor.

The following:
```yaml
packages:
  repositories:
  - name: rhos
    url: 
      repository: "http://download.devel.redhat.com/released/RHEL-7/7.5/Server/x86_64/os/"
      gpg: "http://download.devel.redhat.com/released/RHEL-7/7.5/Server/x86_64/os/RPM-GPG-KEY-redhat-release"
```

produces:
```ini
[rhos]
name = rhos
baseurl = http://download.devel.redhat.com/released/RHEL-7/7.5/Server/x86_64/os/
enabled = 1
gpgkey = http://download.devel.redhat.com/released/RHEL-7/7.5/Server/x86_64/os/RPM-GPG-KEY-redhat-release
gpgcheck = 1
```